### ISSUE TITLE
Tethering: focus support for Nikon D3400

### ISSUE THREAD

**Issue by Baranowski** [darktable-org/darktable#3899]
**Describe the bug**

When shooting tethered with Nikon D3400, Darktable does not control the focus at all. All the captures are done with the same focus. The only way to change the focus is to manually rotate the focus wheel on the lens. The buttons in the darktable to change the focus are disabled.

libghoto2 supports focus with this camera - I have tested it with the `gphoto2` utility.

**To Reproduce**

1. Connect a camera via USB
2. Go to 'Tethering'
3. Click the 'Live View' icon. Verify that you heard the mirror lift up and can see live feed from the camera.
4. Notice that the buttons below the 'Live View' icon are disabled and you cannot click on them.
5. Change focus with the focus wheel on the lens.
6. Capture a picture. Notice that the focus has not changed regardless of the autofocus settings.

**Expected behavior**

After turning on the 'Live View', focus buttons should work. Autofocus mode should be respected.

**Platform (please complete the following information):**

```
$ /opt/darktable/bin/darktable --version
this is darktable 3.1.0+10~gaf8b11a17
copyright (c) 2009-2019 johannes hanika
darktable-dev@lists.darktable.org

compile options:
  bit depth is 64 bit
  normal build
  SSE2 optimized codepath enabled
  OpenMP support enabled
  OpenCL support enabled
  Lua support disabled
  Colord support enabled
  gPhoto2 support enabled
  GraphicsMagick support enabled
  OpenEXR support enabled
```

```
$ uname -sv
Linux #81-Ubuntu SMP Tue Nov 26 12:20:02 UTC 2019
```

(FWiW, I had the same issue with earlier versions of darktable)

**Additional context**

The desired functionality works in `gphoto2`:

```
$ gphoto2 --shell
/> capture-preview
Saving file as capture_preview.jpg
```

(file `capture_preview.jpg` is in focus)

```
 /> set-config manualfocusdrive=-1000
 /> capture-preview
Saving file as capture_preview.jpg
```

(file `capture_preview.jpg` is very blurred)

Autofocus works as well:

```
$ gphoto2 --shell
/> capture-preview 
Saving file as capture_preview.jpg
```

(file `capture_preview.jpg` is out of focus)

```
/> set-config /main/actions/autofocusdrive=1
/> capture-preview 
Saving file as capture_preview.jpg
```

(file `capture_preview.jpg` is in focus)

**Comment by TurboGit** [comment]
The capture support is done by the gphoto2 library. It is not a dt issue.

**Comment by TurboGit** [comment]
I see, this looks like something we can fix.

**Comment by Baranowski** [comment]
Currently DT supports those buttons only for EOS cameras. From `src/common/camera_control.c`:

```
      // TODO: find a more robust way for this, once we find out how to do it with non-EOS cameras
      cam->can_live_view_advanced = cam->can_live_view &&
                                    dt_camctl_camera_property_exists(camctl, cam, "eoszoomposition");
```

And then `src/libs/live_view.c` uses values for the gphoto2 config `manualfocusdrive` that are only valid for Canon cameras, as per the "Manual focus" section of http://gphoto.com/doc/remote/

I won't be working on enabling those buttons, since this feature itself is not very important to me.


Separately from that, it seems that the `autofocusdrive` config from `gphoto2` is not supported by DT at all. At least grepping the source code didn't return anything. I might slowly work on that. (But anyone willing should feel free to take over.)


A great feature that would make the two points above unimportant to me would be for DT to highlight in-focus areas in Live View. I will file a feature request for that and perhaps work on it myself.

**Comment by johnny-bit** [comment]
The advanced tethering for nikon PR #7083 by @vrnhgd got merged so... Can this issue be closed?

**Comment by vrnhgd** [comment]
Yes it can be IMO

**Comment by johnny-bit** [comment]
@Baranowski - I'm closing the issue now. The work by @vrnhgd got merged in and will be part of 3.6 release. I hope that solves the issue and allows greater focus control for Nikon in tethering mode. If there are still some things missing, please open a new issue (you'd have to try the current master though, the PR is not part of 3.4 release. fortunately 3.6 comes out in ~2 weeks)

Thank you @vrnhgd for your work on that :)
